### PR TITLE
make 2022 migration work independently of changes in the future

### PR DESCRIPTION
**Pull request by machisuji**
WP [#50218](https://community.openproject.org/projects/openproject/work_packages/50218/activity)

### CODE REVIEW

File: db/migrate/20220818074150_fix_invalid_journals.rb [hunk 1]
```diff
@@ -34,26 +34,53 @@ def up
...
+
+    Journal.where(journable_type:).where.not(data_type: journal_class.to_s)
+  rescue NameError # the class has been removed in the meantime
+    # rubocop:disable Rails/Output
+    puts "Journable type '#{journable_type}' no longer exists. Removing all its journals."
+    # rubocop:enable Rails/Output
+
+    # the journable type (e.g. WikiContent) doesn't exist anymore
+    # so we can remove all remaining journals for it
+    Journal.where(journable_type:)
```
Comment (oliverguenther) [inline db/migrate/20220818074150_fix_invalid_journals.rb:84]:
```suggestion
    Journal.where(journable_type:).delete_all
```

I guess you forgot the `.delete_all` call?

Comment (ulferts) [inline db/migrate/20220818074150_fix_invalid_journals.rb]: The WikiContent journals are [migrated to journals of the wiki page after this migration is run](https://github.com/opf/openproject/blob/dev/db/migrate/20230322135932_merge_wiki_content_into_page.rb#L114-L127). So the statement that WikiContent doesn't exist anymore is not true for the state where this migration is run in. 

But does this migration really need the instantiated class?

**Review by ulferts — CHANGES_REQUESTED** [review]
Please see my remark in the code

**Review by ulferts — APPROVED** [review]
LGTM

**Pull request merged**